GETTING_STARTED.md

```diff
@@ -0,0 +1,98 @@
+# Introduction
+
+This doc is designed to help beginners get into Melee decompilation (and hopefully contribute). Also, we'll try to answer common questions here and talk about motivations for working on this project.
+
+# Goals of the project
+
+The goal of this repo is to **write C code that, after being compiled, matches the PowerPC assembly 100%**. There's a bit of nuance here though. Let’s first take a look at what we’re working with:
+
+![compilation_diagram](images/compilation_diagram.png)
+
+The details for what produces these different artifacts is out-of-scope. Here’s what some Melee C code looks like:
+
+```c
+void func_8007B8CC(Fighter* fp, HSD_GObj* otherObj) {
+    Fighter* otherFp;
+
+    otherFp = getFighter(otherObj);
+    ((u32 *)fp)[0x466] = (u32) otherObj;
+    fp->x119C_teamUnk = (u8) otherFp->x61B_team;
+    fp->x119D = (u8) otherFp->xC_playerID;
+}
```

Comment:
This is... not clean code at all. Maybe that's what we want, or maybe we should find a prettier snippet to showcase.

Reply:
🤷 . I picked it because it was short and I did it and it's kind of reasonable how it translates to Assembly by looking at it (with only one semantic jump into the `getFighter`). It emphasizes But yeah, a prettier

Reply:
How about this?
https://github.com/doldecomp/melee/blob/4f6e017127ae2f00f359d89fc417d7f77e95f764/src/melee/ft/chara/ftFox/ftFox_SpecialLw.c#L78-L90

Reply:
mm, sure. the assembly is a little long, but should be fine.

Reply:
also, the C code includes the `inline`

Reply:
https://decomp.me/scratch/dDVIh

Reply:
yeah, actually this one is too long. looking around

Reply:
https://decomp.me/scratch/ZN9lq i think this one is good! changing

Reply:
Callbacks are a little advanced, but the asm is simple so it seems fine.